napalm_nxos_ssh/nxos_ssh.py

```diff
@@ -0,0 +1,1513 @@
+            'serial_number': py23_compat.text_type(serial_number),
+            'model': py23_compat.text_type(model),
+            'hostname': py23_compat.text_type(hostname),
+            'fqdn': fqdn,
+            'interface_list': interface_list
+        }
+
+    def get_interfaces(self):
+        """
+        Get interface details.
+
+        last_flapped is not implemented
+
+        Example Output:
+
+        {   u'Vlan1': {   'description': u'',
+                      'is_enabled': True,
+                      'is_up': True,
+                      'last_flapped': -1.0,
+                      'mac_address': u'a493.4cc1.67a7',
+                      'speed': 100},
+        u'Vlan100': {   'description': u'Data Network',
+                        'is_enabled': True,
+                        'is_up': True,
+                        'last_flapped': -1.0,
+                        'mac_address': u'a493.4cc1.67a7',
+                        'speed': 100},
+        u'Vlan200': {   'description': u'Voice Network',
+                        'is_enabled': True,
+                        'is_up': True,
+                        'last_flapped': -1.0,
+                        'mac_address': u'a493.4cc1.67a7',
+                        'speed': 100}}
+        """
+        interfaces = {}
+        command = 'show interface'
+        output = self.device.send_command(command)
+        if not output:
+            return {}
+
+        # Break output into per-interface sections (note, separator text is retained)
+        separator1 = r"^\S+\s+is \S+.*\nadmin state is.*$"
+        separator2 = r"^.* is .*, line protocol is .*$"
+        separators = r"({}|{})".format(separator1, separator2)
+        interface_lines = re.split(separators, output, flags=re.M)
+
+        if len(interface_lines) == 1:
+            msg = "Unexpected output data in '{}':\n\n{}".format(command, interface_lines)
+            raise ValueError(msg)
+
+        # Get rid of the blank data at the beginning
+        interface_lines.pop(0)
+
+        # Must be pairs of data (the separator and section corresponding to it)
+        if len(interface_lines) % 2 != 0:
+            msg = "Unexpected output data in '{}':\n\n{}".format(command, interface_lines)
+            raise ValueError(msg)
+
+        # Combine the separator and section into one string
+        intf_iter = iter(interface_lines)
+        try:
+            new_interfaces = [line + next(intf_iter, '') for line in intf_iter]
+        except TypeError:
+            raise ValueError()
+
+        for entry in new_interfaces:
+            interfaces.update(parse_intf_section(entry))
+
+        return interfaces
+
+    def get_lldp_neighbors(self):
+        results = {}
+        command = 'show lldp neighbors'
+        output = self.device.send_command(command)
+        lldp_neighbors = napalm_base.helpers.textfsm_extractor(
+                            self, 'lldp_neighbors', output)
```

Comment:
Missing template.

```python
TemplateNotImplemented: TextFSM template lldp_neighbors.tpl is not defined under /home/bewing/.virtualenvs/nxos_ssh/lib/python3.6/site-packages/napalm_nxos_ssh/utils/textfsm_templates
```

Reply:
Strange, I didn't run into this earlier. Will have to look into it.

Reply:
@bewing How did you install this library? Did you do a `pip install -e .` or `python setup.py install` or something else?